### PR TITLE
fix(application-package): Fix the return value error of electron.isDevMode API in Windows system in development mode.

### PR DESCRIPTION
**Pull request by zhaomenghuan**
#### What it does

Fix the return value error of electron.isDevMode API in Windows system in development mode.

#### How to test

Run the Theia source code project based on the Windows system, then open DevTools to execute the following code：
```
require('@theia/application-package/lib/environment').environment.electron.isDevMode()
```
image

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: zhaomenghuan <zhaomenghuan02@meituan.com>

### CODE REVIEW

**Review by msujew — COMMENTED** [review]
Hi @zhaomenghuan, thank you for your contribution, do you mind adding testing steps to your PR?

**Comment by zhaomenghuan** [comment]
> Hi @zhaomenghuan, thank you for your contribution, do you mind adding testing steps to your PR?

Ok, I added a test description

**Review by msujew — APPROVED** [review]
I can confirm that the issue exists on `master` and is addressed by this change 👍 

As I can't imagine it leading to regressions on other OS, I'll approve this.

File: dev-packages/application-package/src/environment.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ class ElectronEnv {
         return this.is()
             && typeof process !== 'undefined'
             // `defaultApp` does not exist on the Node.js API, but on electron (`electron.d.ts`).
-            && ((process as any).defaultApp || /node_modules[/]electron[/]/.test(process.execPath)); // eslint-disable-line @typescript-eslint/no-explicit-any
+            && ((process as any).defaultApp || /node_modules[/|\\]electron[/|\\]/.test(process.execPath)); // eslint-disable-line @typescript-eslint/no-explicit-any
```
Comment (paul-marechal) [inline dev-packages/application-package/src/environment.ts:47]: We don't need the `|` symbol here.

**Review by paul-marechal — CHANGES_REQUESTED** [review]

File: dev-packages/application-package/src/environment.ts (hunk 1 above)
Comment (zhaomenghuan) [reply dev-packages/application-package/src/environment.ts]: Thank you，I modified the code.

**Pull request merged**